lib/internal/url.js

```diff
@@ -1265,6 +1265,10 @@ function domainToUnicode(domain) {
   return _domainToUnicode(`${domain}`);
 }
 
+function decodeAuth(str) {
+  return decodeURIComponent(str).replace(':', '%3A').replace('/', '%2F');
```

Comment:
i think you need to use global replace at least ;)

Reply:
@addaleax This is still unaddressed (and, apparently, untested.)